scripts/tosback-import.js

```diff
@@ -0,0 +1,155 @@
+import fsApi from 'fs';
+import path from 'path';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import parser from 'xml2json';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import xPathToCss from 'xpath-to-css';
+import simpleGit from 'simple-git';
+import { assertValid } from './validation/validator.js';
+import serviceSchema from './validation/service.schema.js';
+import { TYPES } from '../src/types.js';
+import fetch from '../src/fetcher/index.js';
+import filter from '../src/filter/index.js';
+
+const fs = fsApi.promises;
+
+const LOCAL_TOSBACK2_REPO = '../../tosdr/tosback2';
+const TOSBACK2_WEB_ROOT = 'https://github.com/tosdr/tosback2';
+const TOSBACK2_RULES_FOLDER_NAME = 'rules';
+
+function getLocalRulesFolder() {
+  return path.join(LOCAL_TOSBACK2_REPO, TOSBACK2_RULES_FOLDER_NAME);
+}
+
+function getGitHubWebUrl(commitHash, filename) {
+  return [
+    TOSBACK2_WEB_ROOT,
+    'blob',
+    commitHash,
+    TOSBACK2_RULES_FOLDER_NAME,
+    filename
+  ].join('/');
+}
+
+async function parseFile(filename) {
+  const data = await fs.readFile(filename);
+  return parser.toJson(data);
+}
+
+function toPascalCase(str) {
+  const lowerCase = str.toLowerCase();
+  return str[0].toUpperCase() + lowerCase.substring(1);
+}
+
+function toType(str) {
+  let found;
+  for (const i in TYPES) {
+    if (TYPES[i].name === str) {
+      found = i;
+      break;
+    }
+  }
+  if (!found) {
+    throw new Error(`Unsupported type: ${str}`);
+  }
+  return found;
+}
+
+async function getFiltered(docObj) {
+  const content = await fetch(docObj.fetch);
+  return filter(content, docObj, []);
+}
+
+async function validDocObj(docObj) {
+  try {
+    const filtered = await getFiltered(docObj);
+    const filtered2 = await getFiltered(docObj);
+    if (filtered !== filtered2) {
+      return false;
+    }
+    return (filtered.length > 100);
```

Comment:
This is only a partial check, the real check is done by running `npm run validate` on the result of this import script.

Reply:
It does seem strange to reimplement parts of validation here. Let's try to reuse the existing code if possible 🙂

Reply:
Put this simple check with arbitrary minimum length `n=100` in a central function, from where `validate.js` can use it too. https://github.com/ambanum/CGUs/pull/71/commits/5f49ea21229bf618f98199cd21bf2f37f82f768c